rudder-core/src/test/scala/com/normation/rudder/services/policies/write/WriteSystemTechniquesTest.scala

```diff
@@ -207,31 +194,48 @@ class WriteSystemTechniquesTest extends Specification with Loggable with BoxSpec
     , systemVariableServiceSpec
   )
 
-  val promiseWritter = new Cf3PromisesFileWriterServiceImpl(
-      techniqueRepository
-    , pathComputer
-    , logNodeConfig
-    , prepareTemplateVariable
-    , new FillTemplatesService()
-    , "/opt/rudder/etc/hooks.d"
-    , Nil
-  )
+
+  /*
+   * We parametrize the output of file writing with a sub-directory name,
+   * so that we can keep each write in it's own directory for debug.
+   */
+  def getPromiseWritter(label: String) = {
+
+    //where the "/var/rudder/share" file is for tests:
+    val SHARE = abstractRoot/s"share-${label}"
+
+    val rootGeneratedPromisesDir = SHARE/"root"
+
+    val pathComputer = new PathComputerImpl(
+        SHARE.getParent + "/"
+      , SHARE.getName
+      , abstractRoot.getAbsolutePath + "/backup"
```

Comment:
it doesn't seems to care. And at least for the generation part, it does not seems to matter.